src/util/threadpool.rs

```diff
@@ -40,11 +41,11 @@ impl<T: Debug> Debug for Task<T> {
 }
 
 impl<T> Task<T> {
-    fn new<F>(id: u64, gid: T, job: F) -> Task<T>
+    fn new<F>(gid: T, job: F) -> Task<T>
         where F: FnOnce() + Send + 'static
     {
         Task {
-            id: id,
+            id: 0,
```

Comment:
When the task is popped from channel, before been pushed into `scheduleQueue` @siddontang